python/lsst/ts/standardscripts/maintel/closed_loop.py

```diff
@@ -0,0 +1,519 @@
+            intended_usage=None if add_remotes else Usages.DryTest,
+            log=self.log,
+        )
+
+        mtcs_usage = None if add_remotes else MTCSUsages.DryTest
+        self.mtcs = MTCS(self.domain, intended_usage=mtcs_usage, log=self.log)
+
+        lsstcam_usage = None if add_remotes else ComCamUsages.StateTransition
+        self.lsstcam = ComCam(
+            self.domain,
+            intended_usage=lsstcam_usage,
+            log=self.log,
+            tcs_ready_to_take_data=self.mtcs.ready_to_take_data,
+        )
+
+        # The following attributes are set via the configuration:
+        self.filter = None
+        self.grating = None
+
+        # exposure time for the intra/extra images (in seconds)
+        self.exposure_time = None
+
+        # end of configurable attributes
+
+    @classmethod
+    def get_schema(cls) -> typing.Dict[str, typing.Any]:
+        schema_yaml = """
+            $schema: http://json-schema.org/draft-07/schema#
+            $id: https://github.com/lsst-ts/ts_standardscripts/auxtel/ClosedLoopCwfs.yaml
+            title: ClosedLoopCwfs v1
+            description: Configuration for ClosedLoopCwfs Script.
+            type: object
+            properties:
+              mode:
+                description: >-
+                    Mode to use for the script. If set to "cwfs", the script will
+                    use the corner wavefront sensors, if set to "fam" the script
+                    will use the full array mode.
+                type: string
+                enum: ["cwfs", "fam"]
+                default: cwfs
+              rotation_angle:
+                type: number
+                default: 0.0
+                description: >-
+                    Rotator angle (deg). This is the angle between the camera
+                    and the boresight.
+              filter:
+                description: Which filter to use when taking intra/extra focal images.
+                type: string
+                default: empty_1
+              grating:
+                description: Which grating to use when taking intra/extra focal images.
+                type: string
+                default: empty_1
+              exposure_time:
+                description: The exposure time to use when taking images (sec).
+                type: number
+                default: 30.
+              dz:
+                description: De-focus to apply when acquiring the intra/extra focal images (mm).
+                type: number
+                default: 0.8
+              extra_focal_offset:
+                description: >-
+                    The additional m2 defocus (mm) to apply for the extra-focal image to be the
+                    same size as the intra. This value is always positive to compensate for
+                    the change in magnification from moving the secondary mirror.
+                type: number
+                default: 0.0011
+              threshold:
```

Comment:
Maybe instead if having a parameter for threshold and one for selecting the DoF, you could merge these two and make the DoF be a "dictionary" where users can specify the threshold. something like this:

```
dof:
    type: object
    description: >-
        Defines the DoF to use and their associated properties.
        Set the threshold to zero to disable a dof.
    properties:
        M2_dz:
            type: object
            properties:
                threshold:
                    type: number
        M2_dx:
            type: object
            properties:
                threshold:
                    type: number
        M2_dy:
            type: object
            properties:
                threshold:
                    type: number
        M2_rx:
            type: object
            properties:
                threshold:
                    type: number
        M2_ry:
            type: object
            properties:
                threshold:
                    type: number
        .
        .
        .
```

Now, there's a question about how to handle defaults. Maybe we should talk about this in a video conf.